miso-service/src/main/java/uk/ac/bbsrc/tgac/miso/service/impl/DefaultKitService.java

```diff
@@ -113,26 +114,46 @@ public long saveKit(Kit kit) throws IOException {
   @Override
   public long saveKitDescriptor(KitDescriptor kitDescriptor) throws IOException {
     authorizationManager.throwIfNonAdmin();
-    if (kitDescriptor.getId() != KitDescriptor.UNSAVED_ID) {
+    if (kitDescriptor.isSaved()) {
       KitDescriptor original = getKitDescriptorById(kitDescriptor.getId());
+      validateChange(kitDescriptor, original);
+      original.setName(kitDescriptor.getName());
+      original.setDescription(kitDescriptor.getDescription());
       original.setVersion(kitDescriptor.getVersion());
       original.setManufacturer(kitDescriptor.getManufacturer());
       original.setPartNumber(kitDescriptor.getPartNumber());
       original.setStockLevel(kitDescriptor.getStockLevel());
       original.setDescription(kitDescriptor.getDescription());
       kitDescriptor = original;
+    } else {
+      validateChange(kitDescriptor, null);
     }
     kitDescriptor.setChangeDetails(authorizationManager.getCurrentUser());
     return kitStore.saveKitDescriptor(kitDescriptor);
   }
 
+  private void validateChange(KitDescriptor kitDescriptor, KitDescriptor beforeChange) throws IOException {
+    List<ValidationError> errors = new ArrayList<>();
+
+    if (isSetAndChanged(KitDescriptor::getName, kitDescriptor, beforeChange)) {
+      if (kitStore.getKitDescriptorByName(kitDescriptor.getName()) != null) {
+        errors.add(new ValidationError("name", "There is already a kit descriptor with this name"));
```

Comment:
What do you think about removing the term "kit descriptor" in user-facing text sicne that distinction was originally for inventory purposes, which is now being handled by RAMEN?

Reply:
We still haven't decided what the interaction between MISO and RAMEN will look like, but it will likely require making the distinction between kits and kit descriptors